plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1701,8 +1706,110 @@ public void testTruncateDecimalTransform()
     @Test
     public void testBucketTransform()
     {
-        String select = "SELECT d_bucket, row_count, d.min AS d_min, d.max AS d_max, b.min AS b_min, b.max AS b_max FROM \"test_bucket_transform$partitions\"";
+        testBucketTransformForType(
+                "DATE",
+                ImmutableList.of(
+                        "DATE '2020-05-19'",
+                        "DATE '2020-06-19'",
+                        "DATE '2020-07-19'",
+                        "DATE '2020-08-19'"),
+                ImmutableList.of(1, 2, 3, 4),
+                ImmutableList.of(
+                        "DATE '2020-09-19'",
+                        "DATE '2020-10-19'",
+                        "DATE '2020-11-19'"),
+                ImmutableList.of(5, 6, 7),
+                "CAST('2020-05-19' AS DATE)",
+                "CAST('2020-08-19' AS DATE)",
+                "CAST('2020-09-19' AS DATE)",
+                "CAST('2020-11-19' AS DATE)");
+
+        testBucketTransformForType(
+                "UUID",
+                ImmutableList.of(
+                        format("UUID '%s'", UUID.fromString("50000000-0000-0000-0000-000000000000")),
+                        format("UUID '%s'", UUID.fromString("60000000-0000-0000-0000-000000000000")),
+                        format("UUID '%s'", UUID.fromString("20000000-0000-0000-0000-000000000000"))),
```

Comment:
use UUID values with few zeros, like `578787df-fad2-4f88-ba4f-243e2d9d5263`

Reply:
that's a good idea